MoleServer.js

```diff
@@ -63,36 +64,50 @@ class MoleServer {
                     message: 'Method not found'
                 }
             };
-        } else {
-            this.currentTransport = transport;
-
-            try {
-                const result = await this.methods[methodName].apply(this.methods, params);
-
-                if (id !==0 && !id) return; // For notifications do not respond. "" means send nothing
-
-                return {
-                    jsonrpc: '2.0',
-                    id,
-                    result: typeof result === 'undefined' ? null : result
-                };
-            } catch (error) {
-                return {
-                    jsonrpc: '2.0',
-                    id,
-                    error: {
-                        code: errorCodes.EXECUTION_ERROR,
-                        message: 'Method has returned error',
-                        data: (error instanceof Error ? error.message : error)
-                    }
-                };
-            }
         }
+
+        this.currentTransport = transport;
+
+        try {
+            const result = await this.methods[methodName].apply(this.methods, params);
+
+            if (id !==0 && !id) return; // For notifications do not respond. "" means send nothing
+
+            return {
+                jsonrpc: '2.0',
+                id,
+                result: typeof result === 'undefined' ? null : result
+            };
+        } catch (error) {
+            return {
+                jsonrpc: '2.0',
+                id,
+                error: {
+                    code: errorCodes.EXECUTION_ERROR,
+                    message: 'Method has returned error',
+                    data: (error instanceof Error ? error.message : error)
+                }
+            };
+        }
+    }
+
+    _isMethodExposed(methodName) {
+        return (
+            this.methods[methodName] &&
+            typeof this.methods[methodName] === 'function' &&
+            methodName !== 'constructor' &&
+            (!methodName.startsWith('_') || INTERNAL_METHODS_NAMES.includes(methodName)) &&
+            this.methods[methodName] !== Object.prototype[methodName]
+        );
+    }
+
+    _handlePing() {
+        return 'pong';
     }
 
     async run() {
         for (const transport of this.transportsToRegister) {
-            this.registerTransport(transport);
+            await this.registerTransport(transport);
```

Comment:
The old issue when after "await moleServer.run()" we cannot be sure that all transports are initialised